### ISSUE TITLE
indent rule broken inside fat-arrow function

### ISSUE THREAD

**Issue by NicholasBoll** [eslint/eslint#7732]
**Tell us about your environment**

* **ESLint Version:** 3.12.0
* **Node Version:** 6.2.0
* **npm Version:** 3.8.9

**What parser (default, Babel-ESLint, etc.) are you using?**
`babel-eslint`

**Please show your full configuration:**

**1**
```
{
  "rules": {
    "indent" : [2, 2, {"ObjectExpression": 1}]
  }
}
```
**2**
```
{
  "rules": {
    "indent" : [2, 2, {"ObjectExpression": "first"}]
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
const lamda = (filterName, display) => {
  return Object.assign({},
    filterName,
    {
      display,
    }
  );
};
```

**What did you expect to happen?**
No error

**What actually happened? Please include the actual, raw output from ESLint.**
**1**
```bash
    4:5   error    Expected indentation of 23 spaces but found 4   indent
    5:7   error    Expected indentation of 25 spaces but found 6   indent
    6:5   error    Expected indentation of 23 spaces but found 4   indent
```
**2**
```bash
    4:5   error    Expected indentation of 23 spaces but found 4   indent
    6:5   error    Expected indentation of 23 spaces but found 4   indent
```

**The following code passes:**
**1**
```js
const lamda = (filterName, display) => {
  return Object.assign({},
    filterName,
                       {
                         display,
                       }
  );
};
```
**2**
```js
const lamda = (filterName, display) => {
  return Object.assign({},
    filterName,
                       {
      display,
                       }
  );
};
```

**Comment by not-an-aardvark** [comment]
Thanks for the report. I was able to reproduce this issue.
